### PR TITLE
DIMS Redux

### PR DESCRIPTION
**Pull request by Aliaksandr-Yanushkevich**
Hello. I've added Redux in DIMS project. Please, check my code. Thanks!
Deploy link https://dims.netlify.app

role | email
-- | --
admin | admin@gmail.com
mentor | mentor@gmail.com
member | member@gmail.com

Default password for all roles: incubator

I suggest to check 'Remember me' field on login page - this action save your data in session storage and get out you of trouble with tediculous filling out form after every page refreshing.

### CODE REVIEW

File: src/App.module.scss [hunk 1]
```diff
@@ -1,8 +1,9 @@
 @import './styleguide/styleguide.scss';
 
 body {
-  font-family: $font;
+  font-family: $font !important;
```
Comment (maximsan) [inline src/App.module.scss:4]: what is the reason to use !important?

File: src/components/TaskPage/MemberList.jsx [hunk 2]
```diff
@@ -1,18 +1,42 @@
...
+    const userTaskId = generateID();
+    let memberTasks = userTasks;
+    if (checked) {
+      if (!usersWithTaskFromDB.includes(id)) {
+        memberTasks = [...memberTasks, { id, taskId, stateId, userTaskId }];
+      }
+      setUsersWithTask([...usersWithTaskLocal, id]);
+    } else {
+      memberTasks = memberTasks.filter((member) => member.id !== id);
+      setUsersWithTask([...usersWithTaskLocal.filter((memberId) => memberId !== id)]);
```
Comment (maximsan) [inline src/components/TaskPage/MemberList.jsx:29]: why do you need spread here ?

File: src/components/common/Header/Header.jsx [hunk 3]
```diff
@@ -1,56 +1,82 @@
...
-import Button from '../../Button/Button';
 import UserBlock from './UserBlock';
+import Account from '../../Account/Account';
+import { showAccountPage } from '../../../redux/reducers/appReducer';
+
+const Header = ({ role, showAccountPage, accountPageIsVisible }) => {
+  const [activeTab, setActiveTab] = useState('1');
+
+  const toggle = (tab) => {
+    if (activeTab !== tab) setActiveTab(tab);
```
Comment (maximsan) [inline src/components/common/Header/Header.jsx:17]: wrap in curly brackets

File: src/redux/reducers/memberPageReducer.js [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+    case ON_CHANGE:
+      const { type, ...rest } = action;
+      return {
+        ...state,
+        ...rest,
+      };
+    case SHOW_MEMBER_PAGE:
+      return { ...state, memberPageIsVisible: action.memberPageIsVisible };
+    case CLEAR_MEMBER_PAGE:
+      return { ...initialState };
```
Comment (maximsan) [inline src/redux/reducers/memberPageReducer.js:61]: you can return just `initialState`

File: src/redux/reducers/membersReducer.js [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+    }
+    dispatch(setMembers(result));
+  });
+};
+
+export const getDirections = () => (dispatch) => {
+  firebaseApi.getDirections().then((result) => {
+    dispatch(setDirections(result));
+  });
+};
```
Comment (maximsan) [inline src/redux/reducers/membersReducer.js:29]: Please, place all you thunks after reducer in all files

File: src/App.js [hunk 6]
```diff
@@ -105,4 +55,10 @@ class App extends Component {
   }
 }
 
-export default App;
+const mapStateToProps = (state) => {
```
[the remark this answers is not in the document]
Comment (Aliaksandr-Yanushkevich) [reply src/App.js:58]: destructured state in all mapStateToProps

File: src/App.js [hunk 7]
```diff
@@ -105,4 +55,10 @@ class App extends Component {
...
+  const { currentTaskId, currentUserId, accountPageIsVisible } = state.app;
+  const { isAuth, role, userId, firstName, lastName } = state.auth;
+  return { currentTaskId, currentUserId, isAuth, role, userId, firstName, lastName, accountPageIsVisible };
+};
+
+export default connect(mapStateToProps, {})(App);
```
[the remark this answers is not in the document]
Comment (Aliaksandr-Yanushkevich) [reply src/App.js:64]: fixed

File: src/components/TaskPage/MemberList.jsx (hunk 2 above)
Comment (Aliaksandr-Yanushkevich) [reply src/components/TaskPage/MemberList.jsx:29]: fixed

File: src/components/common/Header/Header.jsx (hunk 3 above)
Comment (Aliaksandr-Yanushkevich) [reply src/components/common/Header/Header.jsx:17]: fixed

File: src/redux/reducers/memberPageReducer.js (hunk 4 above)
Comment (Aliaksandr-Yanushkevich) [reply src/redux/reducers/memberPageReducer.js:61]: fixed

File: src/redux/reducers/membersReducer.js (hunk 5 above)
Comment (Aliaksandr-Yanushkevich) [reply src/redux/reducers/membersReducer.js:29]: replaced all thunks after reducers and rename all file with it

**Pull request merged**